components/automate-cli/cmd/chef-automate/config.go

```diff
@@ -513,3 +575,146 @@ func getMergerTOMLPath(args []string, infra *AutomteHAInfraDetails, timestamp st
 
 	return tomlFile, nil
 }
+
+//getConfigFromRemoteServer gets the config for remote server using the commands
+func getConfigFromRemoteServer(infra *AutomteHAInfraDetails, remoteType string, config string) (string, error) {
+	sshUser := infra.Outputs.SSHUser.Value
+	sskKeyFile := infra.Outputs.SSHKeyFile.Value
+	sshPort := infra.Outputs.SSHPort.Value
+
+	remoteIP, remoteService := getRemoteType(remoteType, infra)
+	scriptCommands := fmt.Sprintf(config, remoteService)
+	rawOutput, err := ConnectAndExecuteCommandOnRemote(sshUser, sshPort, sskKeyFile, remoteIP, scriptCommands)
+	if err != nil {
+		return "", err
+	}
+	return rawOutput, nil
+}
+
+//getDecodedConfig gets the decoded config from the input
+func getDecodedConfig(input string, remoteService string) (interface{}, error) {
+	if remoteService == "postgresql" {
+		var src PostgresqlConfig
+		if _, err := toml.Decode(cleanToml(input), &src); err != nil {
+			return nil, err
+		}
+		return src, nil
+	}
+	if remoteService == "opensearch" {
+		var src OpensearchConfig
+		if _, err := toml.Decode(cleanToml(input), &src); err != nil {
+			return nil, err
+		}
+		return src, nil
+	}
+
+	return nil, nil
+}
+
+//getConfigForArgsPostgresqlAndOpenSearch gets the requested config from the args provided for postgresql or opensearch
+func getConfigForArgsPostgresqlOrOpenSearch(args []string, remoteService string) (interface{}, error) {
+	pemBytes, err := ioutil.ReadFile(args[0]) // nosemgrep
+	if err != nil {
+		return nil, err
+	}
+	destString := string(pemBytes)
+	if remoteService == "postgresql" {
+		var dest PostgresqlConfig
+		if _, err := toml.Decode(destString, &dest); err != nil {
+			return dest, errors.Errorf(configValid, remoteService)
+		}
+		return dest, nil
+	} else {
+		var dest OpensearchConfig
+		if _, err := toml.Decode(destString, &dest); err != nil {
+			return dest, errors.Errorf(configValid, remoteService)
+		}
+		return dest, nil
+	}
+
+	return nil, nil
+
+}
+
+//isConfigChanged checks if configuration is changed
+func isConfigChanged(src interface{}, dest interface{}) bool {
+	if reflect.DeepEqual(src, dest) {
+		return false
+	}
+	return true
+}
+
+//getExistingAndRequestedConfigForPostgres get requested and existing config for postgresql
+func getExistingAndRequestedConfigForPostgres(args []string, infra *AutomteHAInfraDetails, config string) (PostgresqlConfig, PostgresqlConfig, error) {
+	//Getting Existing config from server
+	var existingConfig PostgresqlConfig
+	var reqConfig PostgresqlConfig
+	srcInputString, err := getConfigFromRemoteServer(infra, "postgresql", config)
```

Comment:
one last thing, declare "postgresql" as a constant and refer to it instead of directly using. Similar thing for open search as well.